src/cgutils.cpp

```diff
@@ -402,26 +402,17 @@ static Type *bitstype_to_llvm(jl_value_t *bt)
         return T_int8;
     if (bt == (jl_value_t*)jl_long_type)
         return T_size;
+    if (bt == (jl_value_t*)jl_float32_type)
```

Comment:
This isn't supporting Float16, this is defining the C ABI conversion layer (which usually either doesn't define `half float`, or defines that it is implicitly converted to `float`). This commit shouldn't have any effect on the compiler float16 intrinsic support.